ocaml/testsuite/tests/typing-unique/unique.ml

```diff
@@ -561,6 +561,21 @@ Error: This function when partially applied returns a once value,
        but expected to be many.
 |}]
 
+let curry : unique_ box -> (unique_ box -> unique_ box) = fun b1 -> function | b2 -> b2
+[%%expect{|
+Line 1, characters 58-87:
+1 | let curry : unique_ box -> (unique_ box -> unique_ box) = fun b1 -> function | b2 -> b2
+                                                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Error: This function when partially applied returns a once value,
+       but expected to be many.
+|}]
+
+(* For n-ary functions, inner functions are not constrained *)
+let no_curry : unique_ box -> (unique_ box -> unique_ box) = fun b1 -> fun b2 -> b2
```

Comment:
This presumably works only because of submoding. That is, `fun b1 -> fun b2 -> b1` would be rejected. Maybe add another test?

Reply:
Yeah the example you gave would be rejected. But I think the test is confusing, since the thing we are testing is not related to what the function returns. I have made them all return `unit`.